l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -60,73 +77,275 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
     /// @dev A mapping L1 token address => user address => the total deposited amount by the user
     mapping(address => mapping(address => uint256)) private __DEPRECATED_totalDepositedAmountPerUser;
 
+    /// @dev The hash of the factory deps, stored here so that the factory deps that each chain provides can be checked when it start the l2 bridge deployment
+    bytes32 internal factoryDepsHash;
+
+    /// @dev A mapping chainId => bridgeProxy. Used to store the bridge proxy's address, and to see if it has been deployed yet.
+    mapping(uint256 => address) public l2BridgeAddress;
+
+    /// @dev A mapping chainId => l2TokenBeacon. Used to store the token beacon proxy's address, and to see if it has been deployed yet.
+    mapping(uint256 => address) public l2TokenBeaconAddress;
+
+    /// @dev A mapping chainId => bridgeProxy. Used to store the bridge proxy's address after deployment has started, but before it has finished.
+    mapping(uint256 => address) internal l2BridgePotentialAddress;
+
+    /// @dev A mapping chainId => l2TokenBeacon. Used to store the token beacon proxy's address after deployment has started, but before it has finished.
+    mapping(uint256 => address) internal l2TokenBeaconPotentialAddress;
+
+    /// @dev we have to record if the bridgeImplTx succeeded
+    mapping(uint256 => bool) internal bridgeImplTxSucceeded;
+
+    /// @dev A mapping chainId => bridgeImplTxHash. Used to check the deploy transaction of the l2Bridge Implementation (which depends on its place in the priority queue).
+    mapping(uint256 => bytes32) internal bridgeImplDeployOnL2TxHash;
+
+    /// @dev A mapping chainId => bridgeProxyTxHash. Used to check the deploy transaction of the l2Bridge Proxy (which depends on its place in the priority queue).
+    mapping(uint256 => bytes32) public bridgeProxyDeployOnL2TxHash;
+
+    /// @dev A mapping L2 _chainId => Batch number => message number => flag
+    /// @dev Used to indicate that L2 -> L1 message was already processed
+    mapping(uint256 => mapping(uint256 => mapping(uint256 => bool))) public isWithdrawalFinalizedShared;
+
+    /// @dev A mapping chainId => keccak256(account, tokenAddress, amount) => L2 deposit transaction hash => true
+    /// @dev Used for saving the number of deposited funds, to claim them in case the deposit transaction will fail
+    mapping(uint256 => mapping(bytes32 => mapping(bytes32 => bool))) public depositHappened;
```

Comment:
Maybe we can do smarter. depositHappened can be a mapping (`chainID` => mapping(L2DepositTxHash) => txDataHash)

Wdyt?

Reply:
very smart